### ISSUE TITLE
data_aug folder....

### ISSUE THREAD

**Issue by Ubiquity123** [michhar/pytorch-yolo-v3-custom#10]
Hey can you please elaborate on the kind of files that resides in data_aug folder. 
I'am new to these things so sorry for asking such foolish questions

**Comment by michhar** [comment]
@Ubiquity123 hello and thank your for your good question.  I don't really expand on this in the README so it is good to ask.

https://github.com/michhar/pytorch-yolo-v3-custom/blob/master/data_aug/data_aug.py

Is a really long file that has, essentially, data transformations that can help augment the data that goes in to the DataLoaders in PyTorch.  Is is a way to add complexity and variability to the training process, hopefully creating a more generalized object detector for your task.

The `data_aug.py` was created by the author of the project from which I forked, but I've added a few.  The most important one is a `class YoloResize(object):` which holds the sizing functions and is the last step before image is fed into the neural network for training and inference.

Each data augmentation class has an `__init__` and `__call__` function.  The `__init__` is basically our instantiator or constructor that sets instance variables.  The `__call__` gets called when the transform is used to augment an image and is actually called by the PyTorch DataLoader during the training or inference pass.
